### PR TITLE
[Editor] Support svg images in the stamp annotation

### PR DESCRIPTION
**Pull request by calixteman**
createImageBitmap doesn't work with svg files (see bug 1841972), so we need to workaround this in using an Image.
When printing/saving we must rasterize the image, hence we get the biggest bitmap as image reference to avoid duplications or poor quality on rendering.

### CODE REVIEW

File: src/display/editor/stamp.js [hunk 1]
```diff
@@ -66,7 +68,11 @@ class StampEditor extends AnnotationEditor {
             this.remove();
             return;
           }
-          ({ bitmap: this.#bitmap, id: this.#bitmapId } = data);
+          ({
+            bitmap: this.#bitmap,
+            id: this.#bitmapId,
+            svg: this.#isSvg,
```
Comment (Snuffleupagus) [inline src/display/editor/stamp.js:74]: Shouldn't this be as follows, looking at the code below in this file?
```suggestion
            isSvg: this.#isSvg,
```

File: src/display/editor/tools.js [hunk 2]
```diff
@@ -103,10 +104,35 @@ class ImageManager {
         image = await response.blob();
       } else {
         data.file = rawData;
-
```
Comment (Snuffleupagus) [inline src/display/editor/tools.js:106]: Nit: Looks unrelated.

File: src/display/editor/tools.js [hunk 3]
```diff
@@ -145,6 +171,14 @@ class ImageManager {
     return this.getFromUrl(data.url);
   }
 
+  getSVGUrl(id) {
```
Comment (Snuffleupagus) [inline src/display/editor/tools.js:174]: Maybe a tiny big easier to read as follows (with fewer upper-case letters in a row)?
```suggestion
  getSvgUrl(id) {
```

File: src/display/editor/tools.js [hunk 4]
```diff
@@ -145,6 +171,14 @@ class ImageManager {
...
+    const data = this.#cache.get(id);
+    if (!data?.isSvg) {
+      return null;
+    }
+    return data.svgrl;
```
Comment (Snuffleupagus) [inline src/display/editor/tools.js:179]: This is always undefined, so I think that you meant the following :-)
```suggestion
    return data.svgUrl;
```

**Comment by calixteman** [comment]
Something else we could try is to make an integration test and at some save the pdf but in using directly the api to get the saved stream, then parse the result and finally check that everything is ok: it'd be a mix of integration test and unit test. Wdyt ?

**Comment by Snuffleupagus** [comment]
> Something else we could try is to make an integration test and at some save the pdf but in using directly the api to get the saved stream, then parse the result and finally check that everything is ok: it'd be a mix of integration test and unit test. Wdyt ?

If it works without too much trouble, that sounds like a good idea!

**Comment by calixteman** [comment]
It's easy to get the bytes buffer from a saved pdf but it's more complex to parse it.
So I just tested that editors and the serialized images have the correct dimensions (which means that the image behind is the correct one).

File: test/integration/stamp_editor_spec.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+          const { width, height } = await getEditorDimensions(page, 1);
+
+          expect(Math.round(parseFloat(width))).toEqual(40);
+          expect(height).toEqual("auto");
+
+          const [bitmap] = await serializeBitmapDimensions(page);
+          // The original size is 80x242 but to increase the resolution when it
+          // is rasterized we scale it up by 96 / 72
+          const ratio = 96 / 72; // PixelsPerInch.PDF_TO_CSS_UNITS
```
Comment (Snuffleupagus) [inline test/integration/stamp_editor_spec.js:115]: Hard-coding this doesn't feel great, and it ought be available as follows.
```suggestion
          const ratio = window.pdfjsLib.PixelsPerInch.PDF_TO_CSS_UNITS;
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, with the comment addressed; thank you!

File: test/integration/stamp_editor_spec.js (hunk 5 above)
Comment (calixteman) [reply test/integration/stamp_editor_spec.js:115]: I looked for something like that and I didn't find it, thx.
